File: BFS/BFS.py
from collections import deque

def BFS(graph, v):
    result = []
    visited = [False] * len(graph)

    visited[v] = True
    queue = deque([v])
	
    while queue:
        vertex = queue.popleft()
        result.append(vertex)

        for curr_v in sorted(graph[vertex]):
            if not visited[curr_v]:
                visited[curr_v] = True
                queue.append(curr_v)
                
    return result

File: BFS/BFS.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>

using namespace std;

vector<int> BFS(const vector<vector<int>>& graph, int v) {
    vector<int> result;
    vector<bool> visited(graph.size(), false);
    queue<int> q;

    visited[v] = true;
    q.push(v);

    while (!q.empty()) {
        int vertex = q.front(); q.pop();
        result.push_back(vertex);

        // vector<int> neighbors = graph[vertex];
        // sort(neighbors.begin(), neighbors.end());

        // for (auto curr_v : neighbors) {

        for (auto curr_v : graph[vertex]) {
            if (!visited[curr_v]) {
                visited[curr_v] = true;
                q.push(curr_v);
            }
        }
    }

    return result;
}
